scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/JsonBucketMetadata.java

```diff
@@ -148,32 +148,12 @@ public void populateDisplayData(Builder builder) {
   }
 
   @Override
-  public boolean isPartitionCompatibleForPrimaryKey(final BucketMetadata o) {
-    if (o == null || getClass() != o.getClass()) return false;
-    JsonBucketMetadata<?, ?> that = (JsonBucketMetadata<?, ?>) o;
-    return getKeyClass() == that.getKeyClass()
-        && keyField.equals(that.keyField)
-        && Arrays.equals(keyPath, that.keyPath);
```

Comment:
This is strange on this one we were checking the path. I suppose this is fine to drop

Reply:
It should be redundant since `keyPath` is [derived deterministically](https://github.com/spotify/scio/blob/main/scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/JsonBucketMetadata.java#L117) from `keyField` !